### PR TITLE
Support path-based authentication with @TestSecurity annotation

### LINKED ISSUE #38298: Mixing auth-mechanizms fails `TestSecurity` tests
### Describe the bug

In case if security is mixed (basic+bearer) accordingly to quarkus doc I have to specify [an authentication mechanizm ](https://quarkus.io/guides/security-authentication-mechanisms#path-specific-authentication-mechanisms)

After that tests with `@TestSecurity` failed.

Looks like the problem is in the transport method - `io.quarkus.test.security.TestHttpAuthenticationMechanism#getCredentialTransport` since it does return null for tests and as a result `findBestCandidateMechanism`  [here ](https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java#L112 )returns baisic auth mechanizm instead of null

### Expected behavior

Test passed

### Actual behavior

Test failed

### How to Reproduce?

To reproduce the issue just replace lines in [application.properties](https://github.com/quarkusio/quarkus/blob/main/integration-tests/elytron-resteasy/src/main/resources/application.properties) file
from
```
%auth-failed-ex-mapper.quarkus.http.auth.permission.basic.paths=/*
%auth-failed-ex-mapper.quarkus.http.auth.permission.basic.policy=authenticated
%auth-failed-ex-mapper.quarkus.http.auth.proactive=false
```
to
```
quarkus.http.test-port=0
quarkus.http.auth.permission.basic.paths=/*
quarkus.http.auth.permission.basic.policy=authenticated
quarkus.http.auth.permission.basic.auth-mechanism=basic
```
and run test [TestSecurityTestCase ](https://github.com/quarkusio/quarkus/blob/main/integration-tests/elytron-resteasy/src/test/java/io/quarkus/it/resteasy/elytron/TestSecurityTestCase.java)

If commented out line
```
quarkus.http.auth.permission.basic.auth-mechanism=basic
```
then everything will work.

The same problem is with bearer mechanizm. 

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

latest

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by michalvavrik**
fixes: #38298

### CODE REVIEW

File: docs/src/main/asciidoc/security-testing.adoc [hunk 1]
```diff
@@ -121,6 +121,38 @@ If it becomes necessary to test security features using both `@TestSecurity` and
 mechanism when none is defined), then Basic Auth needs to be enabled explicitly,
 for example by setting `quarkus.http.auth.basic=true` or `%test.quarkus.http.auth.basic=true`.
 
+=== Path-based authentication
+
+`@TestSecurity` can also be used when xref:security-authentication-mechanisms.adoc#combining-authentication-mechanisms[authentication mechanisms must be combined].
+Example below shows how to select authentication mechanism when path-based authentication is enabled by HTTP Security
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-testing.adoc:127]: I propose to promote the new way of using annotations, so, start with one example where 2 annotations are used only, no policies. And the 2nd example is the current one (but please show the policy requiring form as well)

File: test-framework/security/src/main/java/io/quarkus/test/security/TestSecurity.java [hunk 2]
```diff
@@ -26,5 +28,20 @@
...
+     */
     SecurityAttribute[] attributes() default {};
+
+    /**
+     * Selects authentication mechanism used in a path-based authentication.
+     * If an HTTP Security Policy is used to enable path-based authentication,
+     * then a {@link SecurityIdentity} will only be provided by this annotation if this attribute
+     * matches the 'quarkus.http.auth.permission."permissions".auth-mechanism' configuration property.
+     * Situation is similar when annotations are used to enable path-based authentication for Jakarta REST endpoints.
+     * Set this attribute to 'basic' if an HTTP request to Jakarta REST endpoint annotated with the
```
Comment (sberyozkin) [inline test-framework/security/src/main/java/io/quarkus/test/security/TestSecurity.java:43]:
```suggestion
     * For example, set this attribute to 'basic' if an HTTP request to Jakarta REST endpoint annotated with the
```

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpCredentialTransport.java [hunk 3]
```diff
@@ -54,7 +54,11 @@ public enum Type {
         /**
          * Authorization code, type target is the query 'code' parameter
          */
-        AUTHORIZATION_CODE
+        AUTHORIZATION_CODE,
+        /**
+         * HTTP credential transport is mocked during security testing.
```
Comment (sberyozkin) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpCredentialTransport.java:59]:
```suggestion
         * Reserved for HTTP credential transport used during the security testing with 'io.quarkus.test.security.TestSecurity' annotation.
```

File: docs/src/main/asciidoc/security-testing.adoc (hunk 1 above)
Comment (michalvavrik) [reply docs/src/main/asciidoc/security-testing.adoc]: IMO done, thanks.

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**